### PR TITLE
Add test for getting `DEPLOY_ACCOUNT` transactions

### LINKED ISSUE #456: Add test for `DEPLOY_ACCOUNT` transactions

### PR DESCRIPTION
**Pull request by cptartur**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #456 

* **What is the current behavior?** (You can also link to an open issue here)

### CODE REVIEW

File: starknet_py/tests/e2e/client/conftest.py [hunk 1]
```diff
@@ -207,13 +210,17 @@ def client(request) -> Client:
 
 @pytest_asyncio.fixture(name="prepare_network", scope="module", autouse=True)
 async def fixture_prepare_network(
-    network: str, new_gateway_account_client: AccountClient
+    network: str,
+    new_gateway_account_client: AccountClient,
+    details_of_account_to_be_deployed: AccountToBeDeployedDetails,
```
Comment (cptartur) [inline starknet_py/tests/e2e/client/conftest.py:215]: There's a problem with `details_of_account_to_be_deployed` fixture: it has a scope `module` and even though it generates a random salt and key, it's not rerun for every tests - if one attempts to do multiple deploy_account transactions in one test module, all but first will fail. 

This is problematic and will need to be addressed in separate PR.

File: starknet_py/tests/e2e/client/prepare_net_for_gateway_test.py [hunk 2]
```diff
@@ -67,4 +92,29 @@ async def prepare_net_for_tests(
         block_with_invoke_number=block_with_invoke_number,
         declare_transaction_hash=declare_result.transaction_hash,
         block_with_declare_number=block_with_declare_number,
+        deploy_account_transaction_hash=deploy_account_result.transaction_hash,
+        block_with_deploy_account_hash=block_with_deploy_account_hash,
+        block_with_deploy_account_number=block_with_deploy_account_number,
+    )
+
+
+async def _signed_deploy_account_transaction(
```
Comment (war-in) [inline starknet_py/tests/e2e/client/prepare_net_for_gateway_test.py:101]: Isn't it a duplication of the `deploy_account_transaction` fixture? Maybe there is a chance to reuse it?

File: starknet_py/tests/e2e/client/gateway_test.py [hunk 3]
```diff
@@ -206,3 +207,15 @@ async def test_get_l1_handler_transaction_without_nonce(gateway_client):
 
         assert isinstance(transaction, L1HandlerTransaction)
         assert transaction.nonce is None
+
+
+@pytest.mark.asyncio
+async def test_get_deploy_account_transaction(
+    gateway_client, deploy_account_transaction_hash
```
Comment (war-in) [inline starknet_py/tests/e2e/client/gateway_test.py:214]: Should be marked as `TODO` (`gateway_client` -> `client`)

File: starknet_py/tests/e2e/client/conftest.py (hunk 1 above)
Comment (cptartur) [reply starknet_py/tests/e2e/client/conftest.py:215]: Should be partially resolved now. `details_of_account_to_be_deployed` still isn't re-run, but one can use functions from utils to generate new details.

**Review by war-in — APPROVED** [review]
